Update layout tests for the unified HTTP stack

guoye-zhang opened the pull request:
#### e70acdf22cc938c4c2c3194251d6208171960122
<pre>
Update layout tests for the unified HTTP stack
<a href="https://bugs.webkit.org/show_bug.cgi?id=268193">https://bugs.webkit.org/show_bug.cgi?id=268193</a>
<a href="https://rdar.apple.com/121281240">rdar://121281240</a>

Reviewed by Alex Christensen.

1. &quot;directive-parsing-03&quot; and &quot;source-list-parsing-04&quot; are changed to use a non-ASCII character instead of a disallowed character in HTTP field value. The new HTTP parser sanitizes and converts invalid characters to spaces.
2. Disallowed character is removed from &quot;download_resources_from_invalid_headers&quot; since it already covers other cases.
3. The failure expectation in &quot;resource-request-headers&quot; is removed since the new HTTP stack now passes the test. It is marked as a known failure until the new stack is enabled.

* LayoutTests/http/tests/preload/download_resources_from_invalid_headers-expected.txt:
* LayoutTests/http/tests/preload/resources/nph-invalid_resources_from_header.pl:
* LayoutTests/http/tests/security/contentSecurityPolicy/directive-parsing-03-expected.txt:
* LayoutTests/http/tests/security/contentSecurityPolicy/directive-parsing-03.html:
* LayoutTests/http/tests/security/contentSecurityPolicy/source-list-parsing-04-expected.txt:
* LayoutTests/http/tests/security/contentSecurityPolicy/source-list-parsing-04.html:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac-wk2/http/tests/inspector/network/resource-request-headers-expected.txt:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e70acdf22cc938c4c2c3194251d6208171960122

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37723 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16617 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39964 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40263 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33563 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19262 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13816 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31924 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38290 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13973 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33025 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12218 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12142 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33712 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41522 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34107 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34147 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38044 "Found 1 new test failure: http/tests/security/contentSecurityPolicy/directive-parsing-03.html (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12743 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10285 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36216 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14157 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/33119 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13128 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13471 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on LayoutTests/platform/mac-wk2/TestExpectations:
```diff
@@ -717,8 +720,8 @@ webkit.org/b/141085 http/tests/media/video-query-url.html [ Pass Timeout ]
 
-# Network throttling is currently only enabled on Cocoa platforms.
-http/tests/inspector/network/setEmulatedConditions.html [ Pass ]
+# Network throttling is not supported in the unified HTTP stack.
```
Network throttling is exposed as a feature by Web Inspector, so does this mean that WebKit cannot switch?

aproskuryakov reviewed (changes requested):
Marking as "request changes" until the questions above can be worked out.

guoye-zhang committed:
Update layout tests for the unified HTTP stack
https://bugs.webkit.org/show_bug.cgi?id=268193
rdar://121281240

Reviewed by Alex Christensen.

1. "directive-parsing-03" and "source-list-parsing-04" are changed to use a non-ASCII character instead of a disallowed character in HTTP field value. The new HTTP parser sanitizes and converts invalid characters to spaces.
2. Disallowed character is removed from "download_resources_from_invalid_headers" since it already covers other cases.
3. The failure expectation in "resource-request-headers" is removed since the new HTTP stack now passes the test. It is marked as a known failure until the new stack is enabled.

* LayoutTests/http/tests/preload/download_resources_from_invalid_headers-expected.txt:
* LayoutTests/http/tests/preload/resources/nph-invalid_resources_from_header.pl:
* LayoutTests/http/tests/security/contentSecurityPolicy/directive-parsing-03-expected.txt:
* LayoutTests/http/tests/security/contentSecurityPolicy/directive-parsing-03.html:
* LayoutTests/http/tests/security/contentSecurityPolicy/source-list-parsing-04-expected.txt:
* LayoutTests/http/tests/security/contentSecurityPolicy/source-list-parsing-04.html:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac-wk2/http/tests/inspector/network/resource-request-headers-expected.txt:
